### PR TITLE
added new cue kyverno

### PR DESCRIPTION
**Pull request by pxp928**
Added CUE for creating kyverno Configmap. Also added script to check cosign images before downloading the release.yaml for kyverno.

### CODE REVIEW

**Comment by pxp928** [comment]
Using a template method with cue to create a custom configmap for kyverno:
```
cue export admission-control-verify-image-resources.cue -e template -t key="$USERPUBKEY" --out yaml > $KYVERNO_DIR/admission-control-verify-image-resources.yaml
```

File: platform/vendor/vendor_kyverno.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+KYVERNO_LATEST_STATUS=$?
+KYVERNOPRE_LATEST=`COSIGN_REPOSITORY=$COSIGN_REPOSITORY cosign verify --key $PUBKEY ghcr.io/kyverno/kyvernopre:latest`
+KYVERNOPRE_LATEST_STATUS=$?
+KYVERNOCLI_LATEST=`COSIGN_REPOSITORY=$COSIGN_REPOSITORY cosign verify --key $PUBKEY ghcr.io/kyverno/kyverno-cli:latest`
+KYVERNOCLI_LATEST_STATUS=$?
+
+if [ $KYVERNO_LATEST_STATUS -eq 0 ] && [ $KYVERNOPRE_LATEST_STATUS -eq 0 ] && [ $KYVERNOCLI_LATEST_STATUS -eq 0 ] ; then
+  # Download the release file
+  curl "$KYVERNO_URL" > $KYVERNO_DIR/release.yaml
+  cue export admission-control-verify-image-resources.cue -e template -t key="$USERPUBKEY" --out yaml > $KYVERNO_DIR/admission-control-verify-image-resources.yaml
```
Comment (bradbeck) [inline platform/vendor/vendor_kyverno.sh:28]:
```suggestion
  cue export ${GIT_ROOT}/platform/vendor/admission-control-verify-image-resources.cue -e template -t key="$USERPUBKEY" --out yaml > $KYVERNO_DIR/admission-control-verify-image-resources.yaml
```

File: platform/vendor/admission-control-verify-image-resources.cue [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+		MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEhyQCx0E9wQWSFI9ULGwy3BuRklnt
+		IqozONbbdbqz11hlRJy9c7SG+hdcFl9jE9uE/dwtuwU2MqU9T/cN0YkWww==
+		-----END PUBLIC KEY-----
+		"""
+
+	ghcrio: key.public
+
+	ttlsh: key.public
+	}
+}
```
[the remark this answers is not in the document]
Comment (pxp928) [reply platform/vendor/admission-control-verify-image-resources.cue]: added new line.

File: platform/vendor/vendor_kyverno.sh (hunk 1 above)
Comment (pxp928) [reply platform/vendor/vendor_kyverno.sh]: Changed to `$KYVERNO_DIR/admission-control-verify-image-resources.cue`

**Comment by pxp928** [comment]
Waiting to hear back from Jim about kyverno tagged images not having signatures. Will update as necessary.

**Comment by pxp928** [comment]
Fixed up the deployment script to deploy kyverno to kube cluster.

File: platform/30-kyverno-setup.sh [hunk 3]
```diff
@@ -19,6 +35,20 @@ kubectl create secret generic regcred --type=kubernetes.io/dockerconfigjson --fr
...
+  --type='json' \
+  -p='[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": "--webhooktimeout=15"}]'
+
+kubectl patch deployment \
+  -n kyverno \
+  kyverno \
+  --type='json' \
+  -p='[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": "--imagePullSecrets=regcred"}]'
+
+  kubectl apply -f $KYVERNO_DIR/admission-control-policy
```
Comment (bradbeck) [inline platform/30-kyverno-setup.sh:54]:
```suggestion
kubectl apply -f $KYVERNO_DIR/admission-control-policy.yaml
```

Comment (pxp928) [reply platform/30-kyverno-setup.sh:54]: @bradbeck there are multiple files below that folder that kubectl need to apply. So there is no `.yaml`

Comment (bradbeck) [reply platform/30-kyverno-setup.sh:54]: Ah.

**Comment by mlieberman85** [comment]
Can you just move the files under `admission-control-policy` to somewhere under `/resources/kyverno` ?

The idea is that `platform/vendor` only contains vendor platform dependencies directly, `platform/components` contains our own components and patches to vendor components. For stuff like custom resources it should go under `resources` under something like `resources/kyverno` Open to suggestions as to whether or not it makes sense to do something like resources/admission-policy instead.

We're trying to sort of emulate how a heterogeneous golang based repo would look

**Comment by pxp928** [comment]
> Can you just move the files under `admission-control-policy` to somewhere under `/resources/kyverno` ?
I agree that is a better location for it. Changing it.

**Comment by pxp928** [comment]
@mlieberman85 made the changes!

File: platform/30-kyverno-setup.sh [hunk 4]
```diff
@@ -19,6 +36,20 @@ kubectl create secret generic regcred --type=kubernetes.io/dockerconfigjson --fr
...
+  --type='json' \
+  -p='[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": "--webhooktimeout=15"}]'
+
+kubectl patch deployment \
+  -n kyverno \
+  kyverno \
+  --type='json' \
+  -p='[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": "--imagePullSecrets=regcred"}]'
+
+  kubectl apply -f $KYVERNO_RESOURCE_DIR
```
Comment (bradbeck) [inline platform/30-kyverno-setup.sh:55]:
```suggestion
kubectl apply -f $KYVERNO_RESOURCE_DIR
```

File: platform/vendor/vendor_kyverno.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+KYVERNO_LATEST_STATUS=$?
+KYVERNOPRE_LATEST=`COSIGN_REPOSITORY=$COSIGN_REPOSITORY cosign verify --key $PUBKEY ghcr.io/kyverno/kyvernopre:$KYVERNO_TAG_RELEASE`
+KYVERNOPRE_LATEST_STATUS=$?
+KYVERNOCLI_LATEST=`COSIGN_REPOSITORY=$COSIGN_REPOSITORY cosign verify --key $PUBKEY ghcr.io/kyverno/kyverno-cli:$KYVERNO_TAG_RELEASE`
+KYVERNOCLI_LATEST_STATUS=$?
+
+if [ $KYVERNO_LATEST_STATUS -eq 0 ] && [ $KYVERNOPRE_LATEST_STATUS -eq 0 ] && [ $KYVERNOCLI_LATEST_STATUS -eq 0 ] ; then
+  # Download the release file
+  curl "$KYVERNO_URL" > $KYVERNO_INSTALL_DIR/release.yaml
+  cue export $KYVERNO_RESOURCE_DIR/admission-control-verify-image-resources.cue -e template -t key="$USERPUBKEY" --out yaml > $KYVERNO_RESOURCE_DIR/admission-control-verify-image-resources.yaml
```
Comment (bradbeck) [inline platform/vendor/vendor_kyverno.sh:29]: We might want to put a note in a readme that we now expect `cue` to be available and/or make it part of the setup.

**Review by bradbeck — APPROVED** [review]

File: platform/vendor/vendor_kyverno.sh (hunk 5 above)
Comment (pxp928) [reply platform/vendor/vendor_kyverno.sh:29]: I agree. If that is the tool we are using going forward. It should be part of the setup.

File: platform/30-kyverno-setup.sh (hunk 4 above)
Comment (pxp928) [reply platform/30-kyverno-setup.sh:55]: random space...lol. Removed.

**Review by mlieberman85 — APPROVED** [review]
LGTM, just a follow up should be to update the Readme.

**Pull request merged**